src/Agent.Listener/Configuration/AutoLogonConfigurationManager.cs

```diff
@@ -0,0 +1,258 @@
+    {
+        private ITerminal _terminal;
+        private INativeWindowsServiceHelper _windowsServiceHelper;
+        private IConfigurationStore _store;
+
+        public override void Initialize(IHostContext hostContext)
+        {
+            base.Initialize(hostContext);
+            _terminal = hostContext.GetService<ITerminal>();
+            _windowsServiceHelper = hostContext.GetService<INativeWindowsServiceHelper>();
+            _store = hostContext.GetService<IConfigurationStore>();
+        }
+
+        public void Configure(CommandSettings command)
+        {
+            if (!_windowsServiceHelper.IsRunningInElevatedMode())
+            {
+                Trace.Error("Needs Administrator privileges to configure agent with AutoLogon capability.");
+                Trace.Error("You will need to unconfigure the agent and then re-configure with Administrative rights");            
+                throw new SecurityException(StringUtil.Loc("NeedAdminForAutologonCapability"));
+            }
+
+            string domainName;
+            string userName;
+            string logonAccount;
+            string logonPassword;
+
+            while (true)
+            {   
+                logonAccount = command.GetWindowsLogonAccount(defaultValue: string.Empty, descriptionMsg: StringUtil.Loc("AutoLogonAccountNameDescription"));
+                GetAccountSegments(logonAccount, out domainName, out userName);
+
+                if ((string.IsNullOrEmpty(domainName) || domainName.Equals(".", StringComparison.CurrentCultureIgnoreCase)) && !logonAccount.Contains("@"))
+                {
+                    logonAccount = String.Format("{0}\\{1}", Environment.MachineName, userName);
+                }
+                Trace.Info("LogonAccount after transforming: {0}, user: {1}, domain: {2}", logonAccount, userName, domainName);
+
+                logonPassword = command.GetWindowsLogonPassword(logonAccount);
+                if (_windowsServiceHelper.IsValidCredential(domainName, userName, logonPassword))
+                {
+                    Trace.Info("Credential validation succeeded");
+                    break;
+                }
+                
+                if (command.Unattended)
+                {
+                    throw new SecurityException(StringUtil.Loc("InvalidWindowsCredential"));
+                }
+                    
+                Trace.Error("Invalid credential entered");
+                _terminal.WriteLine(StringUtil.Loc("InvalidWindowsCredential"));
+            }
+
+            bool isCurrentUserSameAsAutoLogonUser = _windowsServiceHelper.HasActiveSession(domainName, userName);                
+            var securityIdForTheUser = _windowsServiceHelper.GetSecurityId(domainName, userName);
+            var regManager = HostContext.GetService<IWindowsRegistryManager>();
+            
+            AutoLogonRegistryManager regHelper = isCurrentUserSameAsAutoLogonUser 
```

Comment:
@TingluoHuang  this comment (https://github.com/Microsoft/vsts-agent/pull/985/files/479eae0578303f4bf9c658182b2c103fe34b224e#r118065264) and other such comments are about the same and posted yesterday. 
If there is anything different/new, please let me know.